src/lib.rs

```diff
@@ -279,6 +279,16 @@ if_everything! {
         peek_bytes(&bytes)
     }
 
+    /// Takes a reference to the first 16 bytes of the total bytes slice and convert it to an array for `peek_bytes` to use.
+    /// Returns None if bytes's length is less than 16.
+    fn take_hint_bytes(bytes: &[u8]) -> Option<&[u8; 16]> {
+        use core::convert::TryInto;
+        bytes.get(0..16)
```

Comment:
Wow this is great, when did this get merged into rust ? (Do you know offhand which rustc this requires ?)

Reply:
`TryInto` trait was stabilized in Rust 1.34